src/coreclr/jit/lower.cpp

```diff
@@ -7801,43 +7801,45 @@ void Lowering::TryRetypingFloatingPointStoreToIntegerStore(GenTree* store)
 //
 // Arguments:
 //    node - GT_AND node of integral type
+//    nextNode - out parameter that represents the 'gtNext' of the given node if the transformation was successful
 //
 // Return Value:
-//    Returns the replacement node if one is created else nullptr indicating no replacement
-GenTree* Lowering::TryLowerAndNegativeOne(GenTreeOp* node)
+//    Returns the true if the transformation was successful; false if it was not.
+bool Lowering::TryLowerAndNegativeOne(GenTreeOp* node, GenTree** nextNode)
 {
     assert(node->OperIs(GT_AND));
 
     if (!varTypeIsIntegral(node))
-        return nullptr;
+        return false;
 
     if (node->gtSetFlags())
-        return nullptr;
+        return false;
 
     if (node->isContained())
-        return nullptr;
+        return false;
 
     GenTree* op2 = node->gtGetOp2();
 
     if (!op2->IsIntegralConst(-1))
-        return nullptr;
+        return false;
 
 #ifndef TARGET_64BIT
     assert(op2->TypeIs(TYP_INT));
 #endif // !TARGET_64BIT
 
     LIR::Use use;
     if (!BlockRange().TryGetUse(node, &use))
-        return nullptr;
+        return false;
```

Comment:
It's worth noting that we can still do the optimization here. It just involves doing:
```cpp
if (BlockRange().TryGetUse(node, &use))
{
    use.ReplaceWith(op1);
}
else
{
    op1->SetUnusedValue();
}

*nextNode = node->gtNext;

BlockRange().Remove(op2);
BlockRange().Remove(node);

return true;
```

Reply:
Ah, interesting, I should try this.

Reply:
It's fine to do this since we've done the work anyway – but note that we expect dead code elimination to remove the unused nodes anyway, so there's no reason to go out of your way to do this if it required something more complicated.